src/design/components/organisms/Modal/index.tsx

```diff
@@ -223,6 +232,30 @@ const ModalItem = ({
     [closeModal]
   )
 
+  useEffectOnce(() => {
```

Comment:
It should be nice to combine the ref hook and the effect hook into a single hook

```ts
const { willUnmountRef } = useWillUnmountRef()
```